app/src/main/java/com/infomaniak/swisstransfer/ui/LaunchActivity.kt

```diff
@@ -88,14 +115,58 @@ class LaunchActivity : ComponentActivity() {
             setClass(this@LaunchActivity, NewTransferActivity::class.java)
             // We need NewMessageActivity to have its standard launchMode in the Manifest
             // in order for FLAG_ACTIVITY_CLEAR_TOP to kill and recreate NewMessageActivity
-            setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_GRANT_READ_URI_PERMISSION)
+            flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_GRANT_READ_URI_PERMISSION
+        }
+    }
+
+    private fun createUploadInProgressIntent(): Intent {
+        val transferType = appSettingsManager.getAppSettings()!!.lastTransferType.toTransferTypeUi()
+        val uploadSession = runBlocking { uploadManager.getLastUpload() }
+
+        return Intent(intent).apply {
+            setClass(this@LaunchActivity, NewTransferActivity::class.java)
+            putExtra(NOTIFICATION_NAVIGATION_KEY, ExternalNavigation.UploadProgress.name)
+            putExtra(TRANSFER_TYPE_KEY, transferType.name)
+            putExtra(TRANSFER_TOTAL_SIZE_KEY, uploadSession?.totalFileSize())
+
+            // We need NewMessageActivity to have its standard launchMode in the Manifest
+            // in order for FLAG_ACTIVITY_CLEAR_TOP to kill and recreate NewMessageActivity
+            flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
+        }
+    }
+
+    private suspend fun createUploadSuccessfulIntent(): Intent {
+        val transferType = appSettingsManager.getAppSettings()!!.lastTransferType.toTransferTypeUi()
+        val lastTransferUuid =
+            applicationContext.lastTransferDataStore.getPreference(LastTransferPreferences.lastTransferUuid)
+        val lastTransferUrl = sharedApiUrlCreator.shareTransferUrl(lastTransferUuid)
+
+        applicationContext.lastTransferDataStore.edit { it.clear() }
+
+        return Intent(intent).apply {
+            setClass(this@LaunchActivity, NewTransferActivity::class.java)
+            putExtra(NOTIFICATION_NAVIGATION_KEY, ExternalNavigation.UploadSuccess.name)
```

Comment:
This `NOTIFICATION_NAVIGATION_KEY` should probably also be renamed.

Maybe something like `EXTERNAL_NAVIGATION_KEY` ?

In doing so, don't forget to also rename the string content itself.

Reply:
Done.